ocaml/typing/solver.ml

```diff
@@ -33,17 +33,37 @@ type 'a error =
     right : 'a
   }
 
-(** Map the function to the list, and returns the first [Error] found;
-    Returns [Ok ()] if no error. *)
-let rec find_error (f : 'x -> ('a, 'b) Result.t) : 'x list -> ('a, 'b) Result.t
-    = function
-  | [] -> Ok ()
-  | x :: rest -> (
-    match f x with Ok () -> find_error f rest | Error _ as e -> e)
-
 module Solver_mono (C : Lattices_mono) = struct
+  type any_morph = Any_morph : ('a, 'b, 'd) C.morph -> any_morph
```

Comment:
I want to annotate this with `[@@unboxed]` for performance but got error because `C.morph` could be a float. @goldfirere How do I say in the interface of `C` that `C.morph` is not a float?

Reply:
You can't. 😢

I don't know of a workaround here.